lib/list_export.rb

```diff
@@ -0,0 +1,67 @@
+class ListExport
+  TYPES = [:media, :feed, :fact_check, :explainer]
+
+  def initialize(type, query, team_id)
+    @type = type
+    @query = query
+    @parsed_query = JSON.parse(@query)
+    @team_id = team_id
+    @team = Team.find(team_id)
+    @feed = Feed.where(id: @parsed_query['feed_id'], team_id: @team_id).last if type == :feed
+    raise "Invalid export type '#{type}'. Should be one of: #{TYPES}" unless TYPES.include?(type)
+  end
+
+  def number_of_rows
+    case @type
+    when :media
+      CheckSearch.new(@query, nil, @team_id).number_of_results
+    when :feed
+      @feed.clusters_count(@parsed_query)
+    when :fact_check
+      @team.filtered_fact_checks(@parsed_query).count
+    when :explainer
+      @team.filtered_explainers(@parsed_query).count
+    end
+  end
+
+  def generate_csv_and_send_email_in_background(user)
+    ListExport.delay.generate_csv_and_send_email(self, user.id)
+  end
+
+  def generate_csv_and_send_email(user)
+    # Convert to CSV
+    csv_string = CSV.generate do |csv|
+      self.export_data.each do |row|
+        csv << row
+      end
+    end
```

Comment:
Just need more details for myself to user `csv` for `do`   and then use `csv << row` so what's the expected value for `csv_string` it will be `csv` array

Reply:
It generates a CSV string, really.